document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/filter/MongoStandardExprNotInRelationalFilterParser.java

```diff
@@ -0,0 +1,26 @@
+package org.hypertrace.core.documentstore.mongo.query.parser.filter;
+
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
+import org.hypertrace.core.documentstore.mongo.query.parser.filter.MongoRelationalFilterParserFactory.MongoRelationalFilterContext;
+
+/**
+ * Note: MongoDB does not have a native "NOT IN" aggregation operator. This parser simulates "NOT
+ * IN" functionality by combining the $not and $in operators.
+ */
+@AllArgsConstructor
+public class MongoStandardExprNotInRelationalFilterParser implements MongoRelationalFilterParser {
```

Comment:
My bad on the naming suggestion earlier. Since we are special casing "NOT_IN", it is no longer a standard one.

```suggestion
public class MongoNotInExprRelationalFilterParser implements MongoRelationalFilterParser {
```